rust/processor/migrations/2024-05-22-200847_add_v1_migration_views/up.sql

```diff
@@ -0,0 +1,319 @@
+CREATE INDEX lm1_ca_ct_at_a_index ON public.fungible_asset_activities USING btree (asset_type, "type", amount)
+CREATE INDEX lm1_ca_oa_ct_at_index ON public.fungible_asset_activities USING btree (owner_address, asset_type, "type", amount)
+CREATE INDEX lm1_ca_oa_igf_index ON public.fungible_asset_activities USING btree (owner_address, is_gas_fee)
+
+-- replace `coin_balances` with `fungible_asset_balances`
+CREATE OR REPLACE VIEW legacy_migration_v1.coin_balances AS
+    SELECT
+        transaction_version,
+        owner_address,
+        -- this is mainly for hashing the coin type for primary key
+        encode(sha256(asset_type::bytea), 'hex') as coin_type_hash,
+        asset_type as coin_type,
+        amount,
+        transaction_timestamp,
+        inserted_at
+    FROM public.fungible_asset_balances
+    WHERE token_standard = 'v1'
+    ;
+CREATE INDEX lm1_cb_tv_oa_ct_index ON public.fungible_asset_balances USING btree (transaction_version, owner_address, asset_type)
+
+-- replace `current_coin_balances` with `current_fungible_asset_balances`
+CREATE OR REPLACE VIEW legacy_migration_v1.current_coin_balances AS
+    SELECT
+        owner_address,
+        substring(asset_type, 64) as coin_type_hash,
+        asset_type as coin_type,
+        amount,
+        transaction_version as last_transaction_version,
+        transaction_timestamp as last_transaction_timestamp,
+        inserted_at
+    FROM public.current_fungible_asset_balances
+    WHERE token_standard = 'v1'
+    ;
+CREATE INDEX lm1_ccb_ct_a_index ON public.current_fungible_asset_balances USING btree (asset_type, amount)
+
+-- replace `token_activities` with `token_activities_v2`
+-- token_activities_v2.token_data_id is 0x prefixed, but token_activities.token_data_id is not. We need to create an index on the substring
+CREATE OR REPLACE VIEW legacy_migration_v1.token_activities AS
+    SELECT
+        tav.transaction_version,
+        event_account_address,
+        -- These were only used for hashing pk in v1 table
+        0 as event_creation_number,
+        0 as event_sequence_number,
+        tdv.collection_id  as collection_data_id_hash,
+        ltrim(tav.token_data_id, '0x') as token_data_id_hash,
+        property_version_v1 AS property_version,
+        cv.creator_address,
+        cv.collection_name,
+        tdv.token_name AS "name",
+        "type" AS transfer_type,
+        from_address,
+        to_address,
+        token_amount,
+        -- These are not columns in v2
+        NULL AS coin_type,
+        NULL AS coin_amount,
+        tav.inserted_at,
+        tav.transaction_timestamp,
+        event_index
+    FROM public.token_activities_v2 tav 
+    JOIN token_datas_v2 tdv 
+    ON tav.token_data_id = tdv.token_data_id AND tav.transaction_version = tdv.transaction_version
+    JOIN collections_v2 cv
+    ON tdv.collection_id = cv.collection_id AND tdv.transaction_version = cv.transaction_version
```

Comment:
i dont think this works, token_activities_v2 has new entries when tokens have activities whereas collections_v2 has new entries when collection properties change

Reply:
i guess the best we can do is joining on the current_ tables